### ISSUE TITLE
Wont install ruby-runtime 0.12

### ISSUE THREAD

**Issue by SteveyPugs** [elvanja/jenkins-gitlab-hook-plugin#19]
I know this isn't related to this but the pre reqs require it so I am not sure if I am missing something or doing it wrong... 

This is what I get:

```
hudson.util.IOException2: Failed to dynamically deploy this plugin
    at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1317)
    at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1116)
    at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
    at java.util.concurrent.FutureTask$Sync.innerRun(Unknown Source)
    at java.util.concurrent.FutureTask.run(Unknown Source)
    at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:104)
    at java.lang.Thread.run(Unknown Source)
Caused by: java.nio.file.FileSystemException: C:\Program Files (x86)\Jenkins\plugins\ruby-runtime\WEB-INF\lib\classes.jar: The process cannot access the file because it is being used by another process.

    at sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
    at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
    at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
    at sun.nio.fs.WindowsFileSystemProvider.implDelete(Unknown Source)
    at sun.nio.fs.AbstractFileSystemProvider.delete(Unknown Source)
    at java.nio.file.Files.delete(Unknown Source)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at hudson.Util.deleteFile(Util.java:239)
    at hudson.Util.deleteRecursive(Util.java:307)
    at hudson.Util.deleteContentsRecursive(Util.java:204)
    at hudson.Util.deleteRecursive(Util.java:298)
    at hudson.Util.deleteContentsRecursive(Util.java:204)
    at hudson.Util.deleteRecursive(Util.java:298)
    at hudson.Util.deleteContentsRecursive(Util.java:204)
    at hudson.Util.deleteRecursive(Util.java:298)
    at hudson.ClassicPluginStrategy.explode(ClassicPluginStrategy.java:423)
    at hudson.ClassicPluginStrategy.createPluginWrapper(ClassicPluginStrategy.java:129)
    at hudson.PluginManager.dynamicLoad(PluginManager.java:413)
    at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1313)
    ... 6 more
```

I am using jenkins v 1.538

**Comment by tlogik** [comment]
And i can confirm that this is still an issue in jenkins when trying to install ruby-runtime.
Windows server 2012.
Jenkins 1.601
still fails when installing ruby-runtime when Jenkins is installed in a path with spaces, i.e. the default of 

```
C:\Program Files (x86)\Jenkins
```

Installing Jenkins in the root of the c drive solved the issue and ruby-runtime could then be installed.

```
C:\Jenkins
```

**Comment by elvanja** [comment]
@TLogik thanks for the feedback!

**Comment by denza** [comment]
Thanks @elvanja

**Comment by batje** [comment]
This is still an issue in 2.41 (on Windows Server 2011)
This seems like a bug to me (not a question) and it would be great if it could be fixed.

**Comment by CScharer** [comment]
Are you F*in kidding me?  So this tool can't deal with spaces.  WTF?

**Comment by malickf** [comment]
this fix works for me : https://stackoverflow.com/questions/44403642/jenkins-plugin-installation-failing

**Comment by tejareddy2** [comment]
help me overcome this issue indetale

**Comment by amit-ezra** [comment]
had the same issue. 
moving Jenkins installation to c:\Jenkins solved it

**Comment by CScharer** [comment]
I too was able to resolve this my simply moving Jenkins to C:\Jenkins.  Don't forget to change any environment variables, etc... when you do so as well though.

**Comment by anh1979** [comment]
Ahh, I tried this and now I cannot connect to my Jenkins from remote (calling 192.168.xx.xx:8080 produces a timeout), locally / on that machine localhost:8080 works, but tells me that my reverse proxy configuration is wrong, which I definitely didn't set... I scanned the config and settings in the Jenkins folder, but I do not find what is going wrong. Any ideas? Help really appreciated.

Forget the question - somehow the windows firewall re-emerged... D'uh!
